pkg/client/cache/delta_fifo.go

```diff
@@ -199,6 +258,7 @@ func (f *DeltaFIFO) queueActionLocked(actionType DeltaType, obj interface{}) err
 		return KeyError{obj, err}
 	}
 	newDeltas := append(f.items[id], Delta{actionType, obj})
```

Comment:
Sorry, I somehow totally missed this comment. But I don't understand what is non-deterministic about this. Getting a non-existing map element returns the zero value but it doesn't affect the map; see https://play.golang.org/p/0_tmN9pysk